File: src/main/java/com/dh/ondot/schedule/api/response/SchedulePreparationResponse.java
package com.dh.ondot.schedule.api.response;

import com.dh.ondot.schedule.domain.Schedule;

public record SchedulePreparationResponse(
        boolean isMedicationRequired,
        String preparationNote
) {
    public static SchedulePreparationResponse from(Schedule s) {
        return new SchedulePreparationResponse(
                s.isMedicationRequired(),
                s.getPreparationNote()
        );
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/api/response/SchedulePreparationResponse.kt
package com.dh.ondot.schedule.api.response

import com.dh.ondot.schedule.domain.Schedule

data class SchedulePreparationResponse(
    val isMedicationRequired: Boolean,
    val preparationNote: String?,
) {
    companion object {
        @JvmStatic
        fun from(s: Schedule): SchedulePreparationResponse {
            return SchedulePreparationResponse(
                isMedicationRequired = s.isMedicationRequired,
                preparationNote = s.preparationNote,
            )
        }
    }
}
